modules/packages/UnorderedAtomics.chpl

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+   .. warning::
+     This module represents work in progress. The API is unstable and likely to
+     change over time.
+
+   This module provides unordered versions of non-fetching atomic operations for
+   all ``int``, ``uint``, and ``real`` types.  Unordered versions of
+   :proc:`~Atomics.add()`, :proc:`~Atomics.sub()`, :proc:`~Atomics.or()`,
+   :proc:`~Atomics.and()`, and :proc:`~Atomics.xor()` are provided. The results
+   of these functions are not visible until task termination or an explicit
+   :proc:`unorderedAtomicFence()`, but they can provide a significant speedup
+   for bulk atomic operations that do not require ordering of operations:
+
+   .. code-block:: chapel
+
+     use UnorderedAtomics;
+
+     const numTasksPerLocale = here.maxTaskPar,
+           iters = 10000;
+
+     var a: atomic int;
+
+     coforall loc in Locales do on loc do
+       coforall 1..numTasksPerLocale do
+         for i in 1..iters do
+           a.unorderedAdd(i); // unordered atomic add
+
+     // no fence required, fenced at task termination
+
+     const itersSum = iters*(iters+1)/2, // sum from 1..iters
+           numTasks = numLocales * numTasksPerLocale;
+     assert(a.read() == numTasks * itersSum);
+
+   It's important to be aware that unordered atomic operations are not
+   consistent with regular atomic operations and updates may not be visible
+   until the task that issued them terminates or they are explicitly fenced
+   with :proc:`unorderedAtomicFence()`.
```

Comment:
Should it mention the fence at the end of a forall ? And mention the fence at task termination in the prose?

Reply:
Task termination fence is already mentioned, but I will add a note about forall

Reply:
Added note about fence at end of forall (and regenerated docs)